tests/compiler/test_type_hinting.py

```diff
@@ -140,3 +140,28 @@ def test_basic_type_hint_attribute_collision(snippetcompiler):
         """,
         "Can not assign a value of type A to a variable of type int (reported in Construct(A) ({dir}/main.cf:8))",
     )
+
+
+def test_type_inference_is_subclass_right_direction_5790(snippetcompiler):
```

Comment:
Did this indeed fail without your patch?

Reply:
It was failing indeed before the patch with 

```term
E                   inmanta.ast.AmbiguousTypeException: Could not determine namespace for type B. 2 possible candidates exists: [__config__::B, test_5790::B]. To resolve this, use the fully qualified name instead of the short name. (reported in a = Construct(A) (/tmp/tmpr23y_be3/main.cf:15:9))

```